_posts/wasmer-go-embedding-1.0.md

```diff
@@ -0,0 +1,534 @@
+various domains that weren't anticipated. After hundreds of thousands
+of installations, it's our great pleasure to introduce the 1.0
+version, fully rewritten to provide a stable and complete API, better
+performance, cross-compilation, two compilers, two engines, and many
+more advanced features!
+
+* [Improved and simplified API](#improved-and-simplified-api)
+* [Compilers and Engines](#compilers-and-engines)
+* [Cross-compilation](#cross-compilation)
+* [WASI](#wasi)
+* [Ready to use on major platforms and
+  architectures](#ready-to-use-on-major-platforms-and-architectures)
+
+## Improved and simplified API
+
+We have entirely rewritten the project with a new, improved API. All
+the WebAssembly externals are now supported, which includes
+`Function`, `Global`, `Memory`, and `Table`. All of them can be used
+as imports or as exports. Well, this is now straighforward.
+
+The `ImportObject` API is now unified and simplified (previously, we
+had 2 distincts API for historical reasons).
+
+An example is worth a thousand words. Let's consider the following
+example that already illustrates many improvements:
+
+```go
+import "github.com/wasmerio/wasmer-go/wasmer"
+
+// Create an engine. It's responsible to drive the compilation and the
+// execution of a WebAssembly module.
+engine := wasmer.NewEngine()
+
+// Create a store, that holds the engine.
+store := wasmer.NewStore(engine)
+
+// Create a new module from the text representation of a WebAssembly
+// (for the sake of simplicity of the example).
+module, _ := wasmer.NewModule(
+	store,
+	[]byte(`
+		(module
+		  ;; We import a math.sum function.
+		  (import "math" "sum" (func $sum (param i32 i32) (result i32)))
+
+		  ;; We export an add_one function.
+		  (func (export "add_one") (param $x i32) (result i32)
+		    local.get $x
+		    i32.const 1
+		    call $sum))
+	`),
+)
+
+// Let's create a new host function for `math.sum`.
+function := wasmer.NewFunction(
+	store,
+
+	// The function signature.
+	wasmer.NewFunctionType(
+		// Parameters.
+		wasmer.NewValueTypes(wasmer.I32, wasmer.I32),
+		// Results.
+		wasmer.NewValueTypes(wasmer.I32)
+	),
+
+	// The function implementation.
+	func(args []wasmer.Value) ([]wasmer.Value, error) {
+		x := args[0].I32()
+		y := args[1].I32()
+
+		return []wasmer.Value{wasmer.NewI32(x + y)}, nil
+	},
+)
+
+// Let's use the new `ImportObject` API…
+importObject := wasmer.NewImportObject()
+
+// … to register the `math.sum` function.
+importObject.Register(
+	"math",
+	map[string]wasmer.IntoExtern{
+		"sum": function,
+	},
+)
+
+// Finally, let's instantiate the module, with all the imports.
+instance, _ := wasmer.NewInstance(module, importObject)
+
+// And let's call the `add_one` function!
+addOne, _ := instance.Exports.GetFunction("add_one")
+
+result, _ := addOne(41)
+
+assert(result, int32(42))
+```
+
+A couple of interesting things can be seen here:
+
+1. There is an `Engine` API; more about this in the next section,
```

Comment:
unclear of if this is a proper use of `;` or if it should be an m-dash or a comma.

Same below with 
```
no longer receive the famous `context` argument; more on that in a
```

not a big deal though. The other use of `;` looks good though

Reply:
I've changed for “—”.